### PR TITLE
P1/refactor/90 user context

### PR DESCRIPTION
**Pull request by netlightebba**

### CODE REVIEW

**Review by netlight-saraandersson — APPROVED** [review]
Awsome :D

**Comment by netlight-johannadagfalk** [comment]
Perhaps wait for three approvals before merging this one so that we all can have a look to understand :D

**Comment by netlight-johannadagfalk** [comment]
> Perhaps wait for three approvals before merging this one so that we all can have a look to understand :D

I´ve started but is not finished!

**Review by netlight-johannadagfalk — APPROVED** [review]
Looks good :D

File: app/assets/main/react_components/helpers/AchievementsHelper.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            userActionsArray: userPerformedActions,
+          };
+        }
+      })
+      /** Filters the null values */
+      .filter((removed) => removed)
+  );
+};
+
+//retunerar ingenting :( )
```
[the remark this answers is not in the document]
Comment (netlightebba) [reply app/assets/main/react_components/helpers/AchievementsHelper.js:49]: Haha, I will fix this! 🙈

File: app/assets/main/react_components/helpers/AchievementsHelper.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+      ...category,
+      userActionsArray: [...category.userActionsArray, ...allUserActions],
+      actionsArray: [...allActionsWithoutUserActions],
+      id: category.id.toString(),
+    };
+  });
+  return res;
+};
+
+export const updateAchievementsOnDrag = (
```
Comment (elinaolsson) [inline app/assets/main/react_components/helpers/AchievementsHelper.js:83]: Is this function only for Drag or is it for Move like one of the previous functions?

File: app/assets/main/react_components/helpers/AchievementsHelper.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      return action.id == movedItem.id ? { ...action, status: true } : action;
+    });
+
+    /** Removes the action from the actionsArray if the action is performed and exists in updatedActionsArray */
+    const updatedActionsArray = result.actionsArray.filter(
+      (action) => action.id != movedItem.climate_action_id
+    );
+
+    /** Combines the arrays to be able to filter out duplicates in the arrays */
+    const combinedArray = [...updatedUserActionsArray, ...updatedActionsArray];
```
Comment (elinaolsson) [inline app/assets/main/react_components/helpers/AchievementsHelper.js:172]: Maybe a more clear name of what is combined, e.g. "combinedActionsArray"?

File: app/assets/main/react_components/helpers/AchievementsHelper.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+    /** Uses uniqwith to be able to search for unique values */
+    const newCombinedArray = uniqWith(combinedArray, filterDuplicatesNames);
```
Comment (elinaolsson) [inline app/assets/main/react_components/helpers/AchievementsHelper.js:175]: Same here, what is combined?

File: app/assets/main/react_components/helpers/AchievementsHelper.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+    /** Updates the result that consists of no duplicates */
+    const updatedResult = {
```
Comment (elinaolsson) [inline app/assets/main/react_components/helpers/AchievementsHelper.js:178]: Might also be clarified, what result?

File: app/assets/main/react_components/helpers/UserActionsHelper.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+export const acceptedUserActions = (inVal) => {
+  return formatedUserActions(inVal)
+    .filter((action) => action.status !== true)
+    .map((action) => ({ ...action }));
+};
+
+export const columnUserActions = (
+  acceptedList,
+  doneList,
```
Comment (elinaolsson) [inline app/assets/main/react_components/helpers/UserActionsHelper.js:16]: Is "done" a word we use on other places? Or is "performed" or "completed" better?

**Review by elinaolsson — APPROVED** [review]
Wow grymt jobbat! 🤩  🥇

**Pull request merged**